cmd/admin-handlers.go

```diff
@@ -336,6 +337,132 @@ func (a adminAPIHandlers) PerfInfoHandler(w http.ResponseWriter, r *http.Request
 	return
 }
 
+// LockEntry .......
+type LockEntry struct {
+	Timestamp  time.Time `json:"time"`       // Timestamp set at the time of initialization.
+	Name       string    `json:"name"`       // Name contains info like bucket, object etc
+	Type       string    `json:"type"`       // Bool whether write or read lock.
+	Source     string    `json:"source"`     // Source which created the lock
+	ServerList []string  `json:"serverlist"` // RPC path of servers issuing the lock.
+	Client     string    `json:"client"`     // RPC path of client claiming lock.
+	ID         string    `json:"id"`         // UID to uniquely identify request of client.
+}
+
+func setLockEntry(l lockRequesterInfo, name, server string) *LockEntry {
+	entry := &LockEntry{Timestamp: l.Timestamp, Name: name, ServerList: []string{server}, Client: l.Node, Source: l.Source, ID: l.UID}
+	if l.Writer {
+		entry.Type = "Write"
+	} else {
+		entry.Type = "Read"
+	}
+	return entry
+}
+
+// LockEntries - To sort the locks
+type LockEntries []LockEntry
+
+func (l LockEntries) Len() int {
+	return len(l)
+}
+
+func (l LockEntries) Less(i, j int) bool {
+	return l[i].Timestamp.After(l[j].Timestamp)
```

Comment:
should be `l[i].Timestamp.Before(l[j].Timestamp)`, if you need descending entries, you can use sort.Reverse() on the sort.Sort() result

Reply:
done